cpp/src/arrow/compute/options.h

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "arrow/compute/type_fwd.h"
+#include "arrow/result.h"
+#include "arrow/status.h"
+#include "arrow/type_fwd.h"
+#include "arrow/util/visibility.h"
+
+namespace arrow {
+namespace compute {
+
+/// \addtogroup compute-functions
+/// @{
+
```

Comment:
A better name for this header is `function_options.h` to communicate better what the options are about and have it close to `function.h` when both are included.

Reply:
And since `api.h` is a catch-all include for end-user's convenience, we should have `#include "arrow/compute/function_options.h"` in there.

Reply:
> A better name for this header is `function_options.h` to communicate better what the options are about and have it close to `function.h` when both are included.

Sure, will do.

> And since `api.h` is a catch-all include for end-user's convenience, we should have `#include "arrow/compute/function_options.h"` in there.

Did you mean in `compute/api.h` (https://github.com/apache/arrow/pull/39312/files#diff-3e8b6be5373f81be5eeab7b9634a24db3884ca8cbd2c32f03db16bb824df5146R37)? If so, it's already there (of course I will change the file name).

Or in `api.h` of arrow root include? If so, I don't think we should include compute stuff in there.

Would you please help to confirm? Appreciate it.

Reply:
`function.h` renamed to `function_options.h`.

Reply:
Sorry. I meant `compute/api.h` and for some reason I didn't notice the include was already there.